### PR TITLE
PPS strings --> InputTag

### PR DESCRIPTION
**Pull request by AndreaBellora**
This PR is only to make the input variables of the producers consistent across the board and get the flexibility provided by InputTags. It has been tested both with the reco and PPS direct simulation workflows.

### CODE REVIEW

**Comment by AndreaBellora** [comment]
Just adding @jan-kaspar @fabferro to the list of people informed of the PR

File: RecoPPS/Local/plugins/CTPPSPixelLocalTrackProducer.cc [hunk 1]
```diff
@@ -119,7 +119,7 @@ CTPPSPixelLocalTrackProducer::~CTPPSPixelLocalTrackProducer() {}
 void CTPPSPixelLocalTrackProducer::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
   edm::ParameterSetDescription desc;
 
-  desc.add<std::string>("label", "ctppsPixelRecHits")
+  desc.add<edm::InputTag>("label", edm::InputTag("ctppsPixelRecHits"))
       ->setComment("label of the RecHits input for the tracking algorithm");
```
Comment (perrotta) [inline RecoPPS/Local/plugins/CTPPSPixelLocalTrackProducer.cc:123]: Strictly speaking, this is not a "label" any more: perhaps "tag"?

Comment (AndreaBellora) [reply RecoPPS/Local/plugins/CTPPSPixelLocalTrackProducer.cc:123]: I agree, it would be definitely clearer to read. I'm updating it.

**Comment by AndreaBellora** [comment]
It's not clear to me why are the checks failing: RecoPPS/Local/interface/RPixClusterToHit.h has not been touched by the PR. Is there anything I should do about it?

**Comment by perrotta** [comment]
> It's not clear to me why are the checks failing: RecoPPS/Local/interface/RPixClusterToHit.h has not been touched by the PR. Is there anything I should do about it?

It has been touched by #32868, merged a few hours ago.
You must rebase on a IB which includes it: I would suggest to wait for CMSSW_11_3_X_2021-02-23-1100, which will likely include it

**Comment by AndreaBellora** [comment]
> 
> 
> > It's not clear to me why are the checks failing: RecoPPS/Local/interface/RPixClusterToHit.h has not been touched by the PR. Is there anything I should do about it?
> 
> It has been touched by #32868, merged a few hours ago.
> You must rebase on a IB which includes it: I would suggest to wait for CMSSW_11_3_X_2021-02-23-1100, which will likely include it

Thank you very much. I'll wait for it and check how to do it exactly.

**Comment by perrotta** [comment]
code-checks

**Comment by perrotta** [comment]
please test
(it was enough retest based on the lates IB, which already contains the files modified by  #32868: no need to rebase this PR)

**Comment by perrotta** [comment]
+1
- Technical: Pixel digis accessed from the CTPPSPixelClusterProducer via InputTag, and not by the string label any more
- Verified that there is no explicit configuration to update anywhere in CMSSW
- Jenkins tests pass with no changes, as it should

**Comment by qliphy** [comment]
+1

**Pull request merged**